cmd/snap-update-ns/system.go

```diff
@@ -72,6 +72,15 @@ func (upCtx *SystemProfileUpdateContext) Assumptions() *Assumptions {
 	if snapName := snap.InstanceSnap(instanceName); snapName != instanceName {
 		as.AddUnrestrictedPaths("/snap/" + snapName)
 	}
+	// Allow snap-update-ns to write to host's /tmp directory. This is
+	// specifically here to allow two snaps to share X11 sockets that are placed
+	// in the /tmp/.X11-unix/ directory in the private /tmp directories provided
+	// by snap-confine. The X11 interface cannot offer a precise permission for
+	// the slot-side snap, as there is no mechanism to convey this information.
+	// As such, provide write access to all of /tmp.
+	as.AddUnrestrictedPaths("/var/lib/snapd/hostfs/tmp")
+	as.AddModeHint("/var/lib/snapd/hostfs/tmp/snap.*", 0700)
+	as.AddModeHint("/var/lib/snapd/hostfs/tmp/snap.*/tmp", 1777)
```

Comment:
Should there be a hint for `/var/lib/snapd/hostfs/tmp/snap.*/tmp/.X11-unix` too?  We'd want it created with permissions allowing normal users to create sockets in it.  It should have `1777` permissions too.

Reply:
Hmm, are you saying that the slot provider will non be a root process? If so then yes, I will amend this section.

Reply:
Done

Reply:
Definitely.  X servers are usually run as a regular user on modern Linux systems.  If its Xorg running on direct hardware, it generally gains access to the hardware via logind (which we can grant access to with `login-session-control`).  If it's Xwayland, there's no special hardware access needed, so again easy to run as a normal user.